packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -865,59 +903,60 @@ export class GarbageCollector implements IGarbageCollector {
-        // last run that were notified via `referenceAdded`.
-        const explicitReferences: string[] = [];
-        for (const [, outboundRoutes] of Object.entries(this.gcDataFromLastRun.gcNodes)) {
-            explicitReferences.push(...outboundRoutes);
-        }
-        this.referencesSinceLastRun.forEach((outboundRoutes: string[]) => {
-            explicitReferences.push(...outboundRoutes);
-        });
+        const getOutboundRoutes = (nodeId: string, routes: string[]): string[] => {
+            // Outbound routes are routes to datastore nodes that come from a parent node.
+            // There are routes added by gc from a datastore's dds to the datastore which were added implicitly
+            return routes.filter((route) => !nodeId.startsWith(route) && isDataStoreNode(route));
+        };
 
-        // Validate that the current reference graph doesn't have references that we are not already aware of. If this
-        // happens, it might indicate data corruption since we may delete objects prematurely.
-        currentReferences.forEach((route: string) => {
-            // Validate references for data stores only. Currently, layers below data stores don't have GC implemented
-            // so there is no guarantee their references will be notified.
-            if (isDataStoreNode(route) && !explicitReferences.includes(route)) {
-                /**
-                 * The following log will be enabled once this issue is resolved:
-                 * https://github.com/microsoft/FluidFramework/issues/8878.
-                 */
-                // We should ideally throw a data corruption error here. However, send an error for now until we have
-                // implemented sweep and have reasonable confidence in the sweep process.
-                // this.mc.logger.sendErrorEvent({
-                //     eventName: "gcUnknownOutboundRoute",
-                //     route,
-                // });
+        const convertGraphToOutboundReferences = (graph: [string, string[]][]) => {
+            const outboundReferences: IGCReference[] = [];
+            for(const [nodeId, routes] of graph) {
+                const outboundRoutes: string[] = getOutboundRoutes(nodeId, routes);
+                const outboundReferencesForNode = outboundRoutes.map((route): IGCReference => {
+                    return { parent: nodeId, route };
+                });
+                outboundReferences.push(...outboundReferencesForNode);
             }
+            return outboundReferences;
+        };
+
+        const currentReferences = convertGraphToOutboundReferences(Object.entries(currentGCData.gcNodes));
+
+        // Missing Explicit References str Current References not in the previous graph and not in explicit references
+        const missingExplicitReferences: IGCReference[] = currentReferences.filter((reference) => {
+            const previousReferences = previousGCData.gcNodes[reference.parent] ?? [];
```

Comment:
This is complicated than it needs to be because of so many conversions. We already have a structure in the format `node -> set of outbound routes` and we can just that here.
This can become something like -
currentGCData.filter(([nodeId, currentOutboundRoutes]) => {
    const previousRoutes = previousGCData.gcNodes[nodeId] ?? [];
    const explicitRoutes = explicitReferences.get(nodeId) ?? [];
    let found = false;
    currentOutboundRoutes.forEach((route) => {
         found = previousRoutes.includes(route) || explicitRoutes.include(route);
    });
    return found;
});

Reply:
The return type of the findMissingExplicitReferences function should be IGarbageCollectionData or a map of node -> outbound routes similar to this.newReferencesSinceLastRun.